midend/booleanKeys.h

```diff
@@ -0,0 +1,58 @@
+#ifndef MIDEND_BOOLEANKEYS_H_
+#define MIDEND_BOOLEANKEYS_H_
+
+#include "ir/ir.h"
+#include "ir/node.h"
+#include "ir/visitor.h"
+
+namespace P4 {
+
+/// Adds a bit<1> cast to all boolean keys. This is because these keys need to be converted into bit
+/// expressions for the control plane. Also replaces all the corresponding entries in the table with
+/// the appropriate expression.
+/// Example:
+/**
+ *
+ * \code{.cpp}
+ *  table t {
+ *    key = { h.a  : exact; }
+ *    const entries = {
+ *           true : action();
+ +        }
+ *  }
+ *  apply {
+ *    t.apply();
+ *  }
+ * \endcode
+ *
+ * is transformed to
+ *
+ * \code{.cpp}
+ *  table t {
+ *    key = { (bit<1<) h.a  : exact; }
+ *    const entries = {
+ *           1w1 : action();
+ +        }
+ *  }
+ *  apply {
+ *    t.apply();
+ *  }
+ * \endcode
+ *
+ * @pre none
+ * @post all boolean table key expressions are replaced with a bit<1> cast. Constant entries are
```

Comment:
any problems with the control-plane API after this?

Reply:
Will check with a dummy commit. There might be issues.

Reply:
https://github.com/p4lang/p4c/blob/e26a49dbf5749ee92585cec9e95ca781d377ebfd/backends/bmv2/common/expression.cpp#L621

This error will fire if we use the pass for the simple switch.